src/ds3.c

```diff
@@ -1337,6 +1443,62 @@ ds3_error* ds3_delete_bucket(const ds3_client* client, const ds3_request* reques
     return _internal_request_dispatcher(client, request, NULL, NULL, NULL, NULL);
 }
 
+ds3_error* ds3_get_objects(const ds3_client* client, const ds3_request* request, ds3_get_objects_response** _response) {
+    ds3_get_objects_response* response;
+    xmlDocPtr doc;
+    xmlNodePtr root;
+    xmlNodePtr child_node;
+    ds3_error* error;
+    GArray* object_array;
+    GByteArray* xml_blob = g_byte_array_new();
+    error = _internal_request_dispatcher(client, request, xml_blob, load_buffer, NULL, NULL);
+    if(error != NULL) {
+        g_byte_array_free(xml_blob, TRUE);
+        return error;
+    }
+
+    doc = xmlParseMemory((const char*) xml_blob->data, xml_blob->len);
+    if(doc == NULL) {
+        char* message = g_strconcat("Failed to parse response document.  The actual response is: ", xml_blob->data, NULL);
+        g_byte_array_free(xml_blob, TRUE);
+        ds3_error* error = _ds3_create_error(DS3_ERROR_INVALID_XML, message);
+        g_free(message);
+        return error;
+    }
+
+    root = xmlDocGetRootElement(doc);
+
+    if(element_equal(root, "Data") == false) {
+        char* message = g_strconcat("Expected the root element to be 'Data'.  The actual response is: ", xml_blob->data, NULL);
+        g_byte_array_free(xml_blob, TRUE);
+        xmlFreeDoc(doc);
+        ds3_error* error = _ds3_create_error(DS3_ERROR_INVALID_XML, message);
+        g_free(message);
+        return error;
+    }
+
+    object_array = g_array_new(FALSE, TRUE, sizeof(ds3_search_object));
+    response = g_new0(ds3_get_objects_response, 1);
+
+    for(child_node = root->xmlChildrenNode; child_node != NULL; child_node = child_node->next) {
+        if(element_equal(child_node, "S3Object") == true) {
+            ds3_search_object object = *_parse_search_object(doc, child_node);
```

Comment:
Is this one of the memory leaks that we were looking at yesterday?  If so is this fix in the other pull request?